core/src/main/java/com/graphhopper/storage/EdgeAccess.java

```diff
@@ -73,12 +73,14 @@ final void setDist(long edgePointer, double distance) {
      * Translates double distance to integer in order to save it in a DataAccess object
      */
     private int distToInt(double distance) {
-        int integ = (int) (distance * INT_DIST_FACTOR);
-        if (integ < 0)
+        if (distance < 0)
```

Comment:
I now check if distance is negative, because when integ is is negative it could also be because it overflowed.

Reply:
The problem is that we do not handle negative integers in getDist yet:

val / INT_DIST_FACTOR

and I prefer having this check as early as possible. See also #1544

Reply:
Ok, due to MAX_DIST an integer overflow should never occur

Reply:
ok maybe you can just adjust this method ?

Reply:
With this code from master: 
```
    private int distToInt(double distance) {
        int integ = (int) (distance * INT_DIST_FACTOR);
        if (integ < 0)
            throw new IllegalArgumentException("Distance cannot be negative: " + distance);
        if (integ >= Integer.MAX_VALUE)
            return Integer.MAX_VALUE;
        return integ;
    }
```
We would get the exception for too large values of distance because integ overflows and becomes negative, that's why I wanted to check if distance>MAX_DIST directly.

Reply:
Ok, yes. It looks that we do not have a test for this. So when I remove "if (integ >= Integer.MAX_VALUE) return Integer.MAX_VALUE;" everything is fine, which I would fix before doing the refactoring.

Reply:
ok this check makes no sense, because integ is an int and cannot have a value larger than Integer.MAX_VALUE. There is `AbstractGraphStorageTester#testSetTooBigDistance_435` which should fail if a too large distance is not reset to Integer.MAX_VALUE.

Reply:
> integ is an int and cannot have a value larger than Integer.MAX_VALUE
Ups! 

> There is `AbstractGraphStorageTester#testSetTooBigDistance_435` which should fail if a too large distance is not reset to Integer.MAX_VALUE

I see. Looks like your propose change makes a lot more sense.

Reply:
Ouch, the test is also buggy and it should be:

```
 @Test
    public void testSetTooBigDistance_435() {
        graph = createGHStorage();

        double maxDist = EdgeAccess.MAX_DIST;
        EdgeIteratorState edge1 = graph.edge(0, 1, maxDist, true);
        assertEquals(maxDist, edge1.getDistance(), 1);

        try {
            EdgeIteratorState e = graph.edge(0, 2, maxDist + 1, true);
            assertEquals(maxDist, e.getDistance(), .1);
        } catch (Exception ex) {
            fail();
        }
    }
```